src/core/org/apache/jmeter/gui/HtmlReportUI.java

```diff
@@ -0,0 +1,282 @@
+        protected void done() {
+            try {
+                reportLaunchButton.setEnabled(true);
+                reportToUser(get());
+            } catch (InterruptedException | ExecutionException exception) {
+                if (LOGGER.isErrorEnabled()) {
+                    LOGGER.error("Error during html report generation : {}", exception.getMessage(), exception);
+                }
+                List<String> reportException = new ArrayList<>();
+                reportException.add(exception.getMessage());
+                reportToUser(reportException);
+            }
+        }
+    }
+
+    private void reportingAddText(String resString) {
+        reportingArea.setText(reportingArea.getText() + resString + "\n");
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        switch (e.getActionCommand()) {
+        case CREATE_REQUEST:
+            try {
+                reportingArea.setText(JMeterUtils.getResString("generate_report_ui.html_report_processing") + "\n");
+                reportLaunchButton.setForeground(Color.orange);
+                new ReportGenerationWorker(reportLaunchButton).execute();
+            } catch (Exception exception) {
+                if (LOGGER.isErrorEnabled()) {
+                    LOGGER.error("Error during html report generation : {}", exception.getMessage(), exception);
+                }
+            }
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("CSV file path {} \n user.properties file path : {} \nOutput directory file path : {}",
+                        csvFilePathTextField.getText(), userPropertiesFilePathTextField.getText(),
+                        outputDirectoryPathTextField.getText());
+            }
+            break;
+        case BROWSE_USER_PROPERTIES:
+            userPropertiesFilePathTextField.setText(showFileChooser(userPropertiesFileButton.getParent(),
+                    userPropertiesFilePathTextField, false, new String[] { ".properties" }));
+            break;
+        case BROWSE_CSV:
+            csvFilePathTextField.setText(showFileChooser(csvFileButton.getParent(), csvFilePathTextField, false,
+                    new String[] { ".jtl", ".csv" }));
+            break;
+        case BROWSE_OUTPUT:
+            outputDirectoryPathTextField.setText(
+                    showFileChooser(outputDirectoryButton.getParent(), outputDirectoryPathTextField, true, null));
+            break;
+        default:
+            break;
+        }
+    }
+
+    void reportToUser(List<String> runResults) {
+        if (runResults.isEmpty()) {
+            reportingAddText(JMeterUtils.getResString(HtmlReportGenerator.HTML_REPORT_SUCCESS));
+            reportLaunchButton.setForeground(Color.green);
+        } else {
+            StringBuilder sb = new StringBuilder();
+            for (String errorString : runResults) {
+                sb.append(errorString + "\n");
+            }
+            reportingAddText(sb.toString());
```

Comment:
Consider using `String.join("\n", runResults)` and rename runResults to runErrors or similar.